src/main/java/org/jabref/gui/openoffice/DetectOpenOfficeInstallation.java

```diff
@@ -51,33 +46,34 @@ public DetectOpenOfficeInstallation(JDialog parent, OpenOfficePreferences prefer
         if (this.checkAutoDetectedPaths(preferences)) {
             future.complete(true);
         } else {
-            init();
+            DefaultTaskExecutor.runInJavaFXThread(() -> init());
             BackgroundTask.wrap(() -> future.complete(autoDetectPaths()))
-                          .onSuccess(x -> SwingUtilities.invokeLater(progressDialog::dispose))
+                          .onSuccess(x -> progressDialog.close())
```

Comment:
I might be mistaken but I think `executeWith` already returns a `Future` which you should be able to pass along.